test/acquisition/test_knowledge_gradient.py

```diff
@@ -213,6 +214,53 @@ def test_evaluate_q_knowledge_gradient(self):
                     val_expected = (mean * weights).sum(-1).mean(0)
                     self.assertTrue(torch.allclose(val, val_expected))
 
+    def test_evaluate_KG(self):
+        options = {
+            "num_restarts": 2,
+            "raw_samples": 4,
+            "num_inner_restarts": 2,
+            "raw_inner_samples": 3,
+        }
+        for (
+            d,
+            q,
+            n,
+            n_f,
+            objective,
+            X_pending,
+            current_value,
+            dtype,
+        ) in itertools.product(
+            (1, 3),
+            (1, 2),
+            (1, 3),
+            (1, 3),
+            (None, GenericMCObjective(objective=lambda Y: Y.norm(dim=-1))),
+            (None, torch.rand(2, 1)),
+            (None, torch.rand(1)),
+            (torch.float, torch.double),
+        ):
+            # basic test
+            bounds = torch.tensor([[0], [1]], device=self.device, dtype=dtype).repeat(
+                1, d
+            )
+            X_pending = X_pending.expand(-1, d) if X_pending is not None else None
+            train_X = torch.rand(n, d, device=self.device, dtype=dtype)
+            train_Y = torch.rand(n, 1, device=self.device, dtype=dtype)
+            model = SingleTaskGP(train_X, train_Y)
```

Comment:
we'll want to use a `MockModel` if we do a lot of different tests here (see comment above). We should probably  do a separate, basic tests that actually uses a real model, just to test this end-to-end, but doing this for all combos will be too slow.